libs/s3/s3_iam_rest_rgw.py

```diff
@@ -0,0 +1,230 @@
+        signature = f"AWS {self._access_key_id}:{signature_bytes.decode('UTF-8')}"
+        return signature
+
+    async def signed_http_request(
+        self, verb: str, path: str, headers: Dict[str, str] = None,
+        query_params: Dict[str, Any] = None, request_params: Dict[str, Any] = None
+    ) -> Tuple[HTTPStatus, Optional[str]]:
+        """
+        Send an S3-signed request.
+        :param verb: HTTP method of the request (GET, POST, etc.).
+        :param path: server's URI to send the request to (e.g. /admin/user).
+        :param headers: HTTP headers of the request.
+        :param query_params: request parameters to be added into the query.
+        :param request_params: request parameters to be added into the body.
+        :returns: HTTP status and the body of the response.
+        """
+
+        # 'content-type' and 'date' headers need to be signed.
+        # If not provided by the user they might be attached by the HTTP framework
+        # at the time of sending making the signature incorrect.
+        # Here headers are manually added in a proper format.
+        if headers is None:
+            headers = {}
+        if 'content-type' not in headers:
+            headers['content-type'] = 'application/x-www-form-urlencoded'
+        headers['date'] = self.http_date()
+        # 'Authorization' header provides the required signature
+        headers['authorization'] = self._generate_signature(verb, headers, path)
+        try:
+            status, body = await self.request(verb, path, headers, query_params, request_params)
+            return status, body
+        except (HttpClientException, asyncio.TimeoutError) as e:
+            reason = str(e)
+            if isinstance(e, asyncio.TimeoutError):
+                reason = "Request timeout"
+            raise S3ClientException(reason) from None
+
+class RGW_API:
+    
+    async def create_user(self,user_params) -> Tuple[HTTPStatus, Dict[str, Any]]:
+        """
+        Illustrate S3Client signed_http_request work.
+        Create IAM user by specifying parameters, HTTP method and path.
+        :returns: HTTP status code and user information as parsed json.
+        """
+
+        rgwcli = S3Client(
+            ACCESS_KEY_ID, SECRET_ACCESS_KEY, HOST, PORT, tls_enabled=False)
+      
+        status, body = await rgwcli.signed_http_request(
+            'PUT', '/admin/user', query_params=user_params)
+        user_info = json.loads(body)
+        return status, user_info
+    
+    async def delete_user(self,user_params) -> Tuple[HTTPStatus, Dict[str, Any]]:
+        """
+        Illustrate S3Client signed_http_request work.
+        Create IAM user by specifying parameters, HTTP method and path.
+        :returns: HTTP status code and user information as parsed json.
+        """
+
+        rgwcli = S3Client(
+            ACCESS_KEY_ID, SECRET_ACCESS_KEY, HOST, PORT, tls_enabled=False)
+  
+        status, user_info = await rgwcli.signed_http_request(
```

Comment:
Can we have provision to switch between http  and https?

Reply:
Here we are are using HTTP requests